hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/DeregistrationOperation.java

```diff
@@ -45,7 +45,11 @@ protected void runInternal() throws Exception {
         EventServiceImpl eventService = (EventServiceImpl) getNodeEngine().getEventService();
         EventServiceSegment segment = eventService.getSegment(getServiceName(), false);
         if (segment != null) {
-            segment.removeRegistration(topic, id);
+            if (id == null) {
```

Comment:
Seems there is a rolling-upgrade compatibility caveat here:
- You have a hazelcast enterprise cluster with members of mixed versions 5.2 + 5.3 (operating at cluster version 5.2)
- a `DeregistrationOperationSupplier` created on 5.3 member might generate `DeregistrationOperation`s with `id == null`
- Such an operation will be executed on 5.2 member and throw NPE because it will try to execute `segment.removeRegistration(topic, null);`

I think the code paths that lead up to `EventService.deregisterAllListeners` must be guarded so that they are only executed when cluster version is at least 5.3. If the code path starts at user API, then probably the right action is to throw `UnsupportedOperationException` if cluster version is < 5.3.